examples/tensorflow-profiling-examples/trainer-hooks-ext/model.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env python
+
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+
+import tensorflow as tf
+from tensorflow.python.training.basic_session_run_hooks import _HookTimer
+
+tf.logging.set_verbosity(tf.logging.INFO)
+
+BUCKET = None  # set from task.py
+PATTERN = 'of'  # gets all files
+TRAIN_STEPS = 10000
+CSV_COLUMNS = 'weight_pounds,is_male,mother_age,plurality,gestation_weeks,key'.split(',')  # noqa: E501
+LABEL_COLUMN = 'weight_pounds'
+KEY_COLUMN = 'key'
+DEFAULTS = [[0.0], ['null'], [0.0], ['null'], [0.0], ['nokey']]
+
+# Define some hyperparameters
+BATCH_SIZE = 512
+N_EMBEDS = 3
+FIRST_LAYER_SIZE = 64
+NUM_LAYERS = 3
+DROPOUT_RATE = 0.1
+LEARNING_RATE = 0.001
+EVAL_INTERVAL = 200
+
+TRANSFORMED_TRAIN_DATA_FILEBASE = 'train_transformed'
+TRANSFORMED_TEST_DATA_FILEBASE = 'test_transformed'
+EXPORTED_MODEL_DIR = 'exported_model_dir'
+
+
+class LambdaTriggerTimer(_HookTimer):
+    """Timer that triggers according to a lambda function that depends on a step."""
+    def __init__(self, steps_lambda):
+        self._last_triggered_step = None
+        self.steps_lambda = steps_lambda
+
+    def should_trigger_for_step(self, step):
+        if self._last_triggered_step == step:
+            return False
+        if self.steps_lambda(step):
+            return True
+
+        return False
+
+    def update_last_triggered_step(self, step):
+        if self._last_triggered_step is None:
+            elapsed_steps = None
+        else:
+            elapsed_steps = step - self._last_triggered_step
+        self._last_triggered_step = step
+        return (None, elapsed_steps)
```

Comment:
Don't use parens where they aren't required. i.e. this should be `return None, elapsed_steps`